heft-plugins/heft-lint-plugin/src/Eslint.ts

```diff
@@ -47,18 +52,56 @@ async function patchTimerAsync(eslintPackagePath: string, timingsMap: Map<string
   timing.time = patchedTime;
 }
 
-function getFormattedErrorMessage(lintMessage: TEslint.Linter.LintMessage): string {
+function getFormattedErrorMessage(
+  lintMessage: TEslint.Linter.LintMessage | TEslintLegacy.Linter.LintMessage
+): string {
   // https://eslint.org/docs/developer-guide/nodejs-api#◆-lintmessage-type
   return lintMessage.ruleId ? `(${lintMessage.ruleId}) ${lintMessage.message}` : lintMessage.message;
 }
 
-export class Eslint extends LinterBase<TEslint.ESLint.LintResult> {
-  private readonly _eslintPackage: typeof TEslint;
-  private readonly _linter: TEslint.ESLint;
+interface IExtendedEslintConfig extends TEslint.Linter.Config {
+  // https://github.com/eslint/eslint/blob/d6fa4ac031c2fe24fb778e84940393fbda3ddf77/lib/config/config.js#L264
+  toJSON: () => object;
+  __originalToJSON: () => object;
+}
+
+function patchedToJSON(this: IExtendedEslintConfig): object {
+  let programs: TTypescript.Program[] | undefined;
+  if (this.languageOptions?.parserOptions?.programs) {
+    programs = this.languageOptions.parserOptions.programs;
+    delete this.languageOptions.parserOptions.programs;
```

Comment:
Can we set this to `undefined` instead? The `delete` operator is a guaranteed deopt of property accesses from the object, and JSON serialization ignores properties with a value of `undefined`.